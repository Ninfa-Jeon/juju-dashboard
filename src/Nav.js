import React from 'react';
import {Link} from 'react-router-dom'

function Nav() {
  return (
    <div>
      <ul>
        <li><Link to="/">Models</Link></li>
        <li><Link to="/clouds">Clouds</Link></li>
        <li><Link to="/kubernetes">Kubernetes</Link></li>
        <li><Link to="/controllers">Controllers</Link></li>
        <li><Link to="/usage">Usage</Link></li>
        <li><Link to="/logs">Logs</Link></li>
      </ul>
    </div>
  );
}

export default Nav;
