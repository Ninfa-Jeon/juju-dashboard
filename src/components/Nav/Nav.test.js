import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

import Nav from "./Nav";

describe("Nav", () => {
  const wrapper = mount(
    <Router>
      <Nav />
    </Router>
  );
  it("renders without crashing and matches snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
