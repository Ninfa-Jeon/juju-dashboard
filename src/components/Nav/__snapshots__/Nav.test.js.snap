// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders without crashing and matches snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Nav>
      <div>
        <header
          className="p-navigation"
          id="navigation"
        >
          <div
            className="p-navigation__banner"
          >
            <div
              className="p-navigation__logo"
            >
              <a
                className="p-navigation__link"
                href="/"
              >
                <img
                  alt="JAAS logo"
                  className="p-navigation__image"
                  src="https://assets.ubuntu.com/v1/a9e0ed4a-jaas-logo1.svg"
                  width="95"
                />
              </a>
            </div>
            <a
              className="p-navigation__toggle--open"
              href="#navigation"
              title="menu"
            >
              Menu
            </a>
            <a
              className="p-navigation__toggle--close"
              href="#navigation-closed"
              title="close menu"
            >
              Close menu
            </a>
          </div>
          <nav
            className="p-navigation__nav"
          >
            <span
              className="u-off-screen"
            >
              <a
                href="#main-content"
              >
                Jump to main content
              </a>
            </span>
            <ul
              className="p-navigation__links"
              role="menu"
            >
              <li
                className="p-navigation__link is-selected"
                role="menuitem"
              >
                <a
                  className="p-link--external"
                  href="https://jaas.ai/jaas"
                >
                  About
                </a>
              </li>
              <li
                className="p-navigation__link"
                role="menuitem"
              >
                <a
                  className="p-link--external"
                  href="https://discourse.jujucharms.com/"
                >
                  Discourse
                </a>
              </li>
              <li
                className="p-navigation__link"
                role="menuitem"
              >
                <a
                  className="p-link--external"
                  href="https://jaas.ai/docs"
                >
                  Docs
                </a>
              </li>
              <li
                className="p-navigation__link"
                role="menuitem"
              >
                <a
                  className="p-link--external"
                  href="#_"
                >
                  Charmhub
                </a>
              </li>
              <li
                className="p-navigation__link"
                role="menuitem"
              >
                <Link
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Settings
                  </a>
                </Link>
              </li>
            </ul>
          </nav>
        </header>
      </div>
    </Nav>
  </Router>
</BrowserRouter>
`;
